support/cas-server-support-pm-webflow/src/main/java/org/apereo/cas/pm/web/flow/PasswordManagementWebflowConfigurer.java

```diff
@@ -62,23 +72,32 @@ private void createAccountStatusViewStates(final Flow flow) {
                 CasWebflowConstants.VIEW_ID_INVALID_AUTHENTICATION_HOURS);
         createViewState(flow, CasWebflowConstants.VIEW_ID_ACCOUNT_LOCKED, CasWebflowConstants.VIEW_ID_ACCOUNT_LOCKED);
         createViewState(flow, CasWebflowConstants.VIEW_ID_ACCOUNT_DISABLED, CasWebflowConstants.VIEW_ID_ACCOUNT_DISABLED);
-        createEndState(flow, CasWebflowConstants.STATE_ID_PASSWORD_UPDATE_SUCCESS, CasWebflowConstants.VIEW_ID_PASSWORD_UPDATE_SUCCESS);
+        createViewState(flow, CasWebflowConstants.STATE_ID_PASSWORD_UPDATE_SUCCESS, CasWebflowConstants.VIEW_ID_PASSWORD_UPDATE_SUCCESS);
 
         if (casProperties.getAuthn().getPm().isEnabled()) {
             configurePasswordResetFlow(flow, CasWebflowConstants.VIEW_ID_EXPIRED_PASSWORD);
             configurePasswordResetFlow(flow, CasWebflowConstants.VIEW_ID_MUST_CHANGE_PASSWORD);
-            createTransitionForState(flow.getTransitionableState(CasWebflowConstants.VIEW_ID_SHOW_AUTHN_WARNING_MSGS),
-                    CasWebflowConstants.TRANSITION_ID_CHANGE_PASSWORD, CasWebflowConstants.VIEW_ID_MUST_CHANGE_PASSWORD);
+            configureOptionalPasswordChange(flow);
             createPasswordResetFlow();
         } else {
             createViewState(flow, CasWebflowConstants.VIEW_ID_EXPIRED_PASSWORD, CasWebflowConstants.VIEW_ID_EXPIRED_PASSWORD);
             createViewState(flow, CasWebflowConstants.VIEW_ID_MUST_CHANGE_PASSWORD, CasWebflowConstants.VIEW_ID_MUST_CHANGE_PASSWORD);
         }
     }
 
+    private void configureOptionalPasswordChange(final Flow flow) {
+        final TransitionableState warningState = flow.getTransitionableState(CasWebflowConstants.VIEW_ID_SHOW_AUTHN_WARNING_MSGS);
+        warningState.getEntryActionList().add(createEvaluateAction(
```

Comment:
nitpicking: wrapping seems needless to me.

Reply:
Yeah, during my first few attempts that was much longer. I'll move it back inline.